WalletWasabi.Fluent/Helpers/FileDialogHelper.cs

```diff
@@ -6,130 +6,175 @@
+		AppleUniformTypeIdentifiers = new[] { "public.text" },
+		MimeTypes = new[] { "text/plain" }
+	};
 
-	private static SaveFileDialog CreateSaveFileDialog(string title, IEnumerable<string> filterExtTypes, string? directory = null)
+	private static FilePickerFileType Psbt { get; } = new("PSBT files")
 	{
-		var sfd = new SaveFileDialog
-		{
-			DefaultExtension = filterExtTypes.FirstOrDefault(),
-			Title = title,
-			Directory = directory
-		};
+		Patterns = new[] { "*.psbt" },
+		MimeTypes = new[] { "*/*" }
+	};
 
-		return sfd;
-	}
+	private static FilePickerFileType Txn { get; } = new("TXN files")
+	{
+		Patterns = new[] { "*.txn" },
+		MimeTypes = new[] { "*/*" }
+	};
 
-	private static async Task<string?> GetDialogResultAsync(OpenFileDialog ofd)
+	private static FilePickerFileType Png { get; } = new("PNG files")
 	{
-		if (Application.Current?.ApplicationLifetime is IClassicDesktopStyleApplicationLifetime lifetime &&
-			lifetime.MainWindow is { })
+		Patterns = new[] { "*.png" },
+		AppleUniformTypeIdentifiers = new[] { "public.png" },
+		MimeTypes = new[] { "image/png" }
+	};
+
+	private static IStorageProvider? GetStorageProvider()
+	{
+		if (Application.Current?.ApplicationLifetime is IClassicDesktopStyleApplicationLifetime { MainWindow: { } window })
 		{
-			var selected = await ofd.ShowAsync(lifetime.MainWindow);
+			return window.StorageProvider;
+		}
 
-			return selected?.FirstOrDefault();
+		if (Application.Current?.ApplicationLifetime is ISingleViewApplicationLifetime { MainView: { } mainView })
+		{
+			var visualRoot = mainView.GetVisualRoot();
+			if (visualRoot is TopLevel topLevel)
+			{
+				return topLevel.StorageProvider;
+			}
 		}
 
 		return null;
 	}
 
-	private static async Task<string?> GetDialogResultAsync(SaveFileDialog sfd)
+	private static List<FilePickerFileType> GetFilePickerFileTypes(string[] filterExtTypes)
 	{
-		if (Application.Current?.ApplicationLifetime is IClassicDesktopStyleApplicationLifetime lifetime &&
-			lifetime.MainWindow is { })
+		var fileTypeFilters = new List<FilePickerFileType>();
+
+		foreach (var fileType in filterExtTypes)
 		{
-			return await sfd.ShowAsync(lifetime.MainWindow);
+			switch (fileType)
+			{
+				case "*":
+					{
+						fileTypeFilters.Add(All);
+						break;
+					}
+				case "json":
+					{
+						fileTypeFilters.Add(Json);
+						break;
+					}
+				case "txt":
+					{
+						fileTypeFilters.Add(Text);
+						break;
+					}
+				case "psbt":
+					{
+						fileTypeFilters.Add(Psbt);
+						break;
+					}
+				case "txn":
+					{
+						fileTypeFilters.Add(Txn);
+						break;
+					}
+				case "png":
+					{
+						fileTypeFilters.Add(Png);
+						break;
+					}
+			}
 		}
 
-		return null;
+		return fileTypeFilters;
 	}
 
-	private static OpenFileDialog CreateOpenFileDialog(string title, string? directory = null)
+	public static async Task<IStorageFile?> OpenFileAsync(string title, string[] filterExtTypes, string? initialFileName = null, string? directory = null)
```

Comment:
Remove this unused parameter?

```suggestion
	public static async Task<IStorageFile?> OpenFileAsync(string title, string[] filterExtTypes, string? directory = null)
```